azure/converters/image.go

```diff
@@ -49,16 +49,40 @@ func mpImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
 	}, nil
 }
 
-func sigImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
-	imageID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s",
-		image.SharedGallery.SubscriptionID,
-		image.SharedGallery.ResourceGroup,
-		image.SharedGallery.Gallery,
-		image.SharedGallery.Name,
-		image.SharedGallery.Version)
+func computeImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
+	idTemplate := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s"
```

Comment:
the versions part is optional, even though it supports putting /versions/latest, the recommded way is to skip the versions part and just put the image-definition-id for using the "latest"

Reply:
`Version` is right now a _required_ parameter on CAPZ API, hence the formatting. I think in CAPZ case it is beneficial to always require an image version from the user instead of doing that implicitly.

But, given that we introduce a new block for compute galleries, we have a chance to reconsider it. I'll leave that up to maintainers.

Reply:
sounds good

Reply:
I consider it resolved then.